web/main.go

```diff
@@ -180,7 +218,7 @@ func (s *server) authUser(r *http.Request) (user *pb.User, errFunc httpErrorWrit
 		return nil, badRequest, errors.Wrap(err, "failed to decode cookie")
 	}
 
-	userResp, err := s.getUser(userID)
+	userResp, err := s.getUser(trace.NewContext(ctx, span.NewChild("retrieve_user")), userID)
```

Comment:
Actually the code got a ton more clearer when I stopped doing

    ctx := r.Context()
    span := trace.FromContext(ctx)
    childMethod(trace.NewContext(ctx, span.NewChild("foo"))

and just pass the ctx down to the child method. Thanks!